### PR TITLE
update: reading metadata from shared record list

### PR DESCRIPTION
**Pull request by draganaurosgrbic**

### CODE REVIEW

File: src/adiak.c [hunk 1]
```diff
@@ -44,11 +35,21 @@ typedef struct record_list_t {
...
+   int use_mpi;
+   record_list_t **shared_record_list;
+} adiak_t;
+
 static adiak_t *adiak_config;
 static volatile adiak_tool_t **tool_list;
 
 static adiak_tool_t *local_tool_list;
-adiak_t adiak_public = { ADIAK_VERSION, ADIAK_VERSION, 0, 1, &local_tool_list, 0 };
+adiak_t adiak_public = { ADIAK_VERSION, ADIAK_VERSION, 0, 1, &local_tool_list, 0, 0 };
```
Comment (daboehme) [inline src/adiak.c:52]: Let's make this NULL

Comment (draganagrbicllnl) [reply src/adiak.c]: @daboehme done

**Comment by daboehme** [comment]
Looks like the code now requires `adiak_fini` to be called first for `adiak_list_namevals` to work. That's an issue since there are cases where we flush Caliper before finalizing Adiak and that should still work. I think we could just get rid of the static `record_list` variable altogether and only use the pointer in `adiak_public` to make it work.

**Comment by draganagrbicllnl** [comment]
@daboehme I updated the code so it uses only the shared list inside `adiak_public` (static `record_list` is removed)

**Comment by daboehme** [comment]
Superseded by #28

**Pull request closed without merging**